### PR TITLE
[Keyboard] Add Buzzard support

### PR DESCRIPTION
**Pull request by crehmann**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/buzzard/keymaps/crehmann/features/haptic_utils.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            break;
+    }
+
+    return false;
+}
+#endif
+
+void process_layer_pulse(layer_state_t state) {
+#ifdef HAPTIC_ENABLE
+    switch (biton32(state)) {
```
Comment (drashna) [inline keyboards/buzzard/keymaps/crehmann/features/haptic_utils.c:24]:
```suggestion
    switch (get_highest_layer(state)) {
```

File: keyboards/buzzard/rev1/rev1.c [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                                                   0x08, 0x07, 0x07, 0x03, 0x01, 0x01, 0x00, 0x00, 0x03, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+        oled_write_raw_P(buzzard_log, sizeof(buzzard_log));
+    }
+    return false;
+}
+#endif
+
+__attribute__((weak)) layer_state_t layer_state_set_user(layer_state_t state) {
+    switch (biton32(state)) {
```
Comment (drashna) [inline keyboards/buzzard/rev1/rev1.c:100]:
```suggestion
    switch (get_highest_layer(state)) {
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/buzzard/rules.mk [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+CONSOLE_ENABLE = no        	# Console for debug
+COMMAND_ENABLE = no        	# Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no      	# Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no           	# USB Nkey Rollover
+BACKLIGHT_ENABLE = no      	# Enable keyboard backlight functionality
+AUDIO_ENABLE = no          	# Audio output
+SPLIT_KEYBOARD = yes       	# Split common
+LTO_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/buzzard/rules.mk:22]:
```suggestion
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
SPLIT_KEYBOARD = yes
LTO_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/buzzard/buzzard.h [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// Copyright 2021 Christoph Rehmann (crehmann)
+// SPDX-License-Identifier: GPL-2.0-or-later
+#pragma once
+
+#include "quantum.h"
+
+#if defined(KEYBOARD_buzzard_rev1)
+#include "rev1.h"
```
Comment (drashna) [inline keyboards/buzzard/buzzard.h:8]: Nitpick: 
```suggestion
#    include "rev1.h"
```

File: keyboards/buzzard/rev1/rev1.h [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    { L15,   L14,   L13,   L12,   L11,   L10   }, \
+    { L27,   L26,   L25,   L24,   L23,   L22   }, \
+    { L36,   L35,   L34,   KC_NO, KC_NO, KC_NO }, \
+    { R05,   R06,   R07,   R08,   R09,   KC_NO }, \
+    { R16,   R17,   R18,   R19,   R20,   R21   }, \
+    { R28,   R29,   R30,   R31,   R32,   R33   }, \
+    { R37,   R38,   R39,   KC_NO, KC_NO, KC_NO }, \
+}
+
+#define LAYOUT_split_3x6_3 LAYOUT
```
Comment (noroadsleft) [inline keyboards/buzzard/rev1/rev1.h:54]: This isn't needed.

```suggestion
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/buzzard/rev1/rev1.h (hunk 5 above)
Comment (crehmann) [reply keyboards/buzzard/rev1/rev1.h]: Thanks good catch! I removed it

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**